lib/Parse/Lexer.cpp

```diff
@@ -1788,13 +1791,26 @@ static void validateMultilineIndents(const Token &Str,
 
 /// Emit diagnostics for single-quote string and suggest replacement
 /// with double-quoted equivalent.
-static void diagnoseSingleQuoteStringLiteral(const char *TokStart,
-                                             const char *TokEnd,
-                                             DiagnosticEngine *D) {
+///
+/// Or, if we're in experimental regex mode, we will emit a custom
+/// error message instead, determined by the Swift library.
+void Lexer::diagnoseSingleQuoteStringLiteral(const char *TokStart,
+                                             const char *TokEnd) {
   assert(*TokStart == '\'' && TokEnd[-1] == '\'');
-  if (!D)
+  if (!Diags) // or assert?
     return;
 
+  auto startLoc = Lexer::getSourceLoc(TokStart);
+  auto endLoc = Lexer::getSourceLoc(TokEnd);
+
+  if (LangOpts.EnableExperimentalRegex) {
+    auto copy = std::string(TokStart, TokEnd-TokStart);
+    auto msg = experimental_regex_strawperson(copy.c_str());
+    if (msg != nullptr) {
+      Diags->diagnose(startLoc, diag::lex_experimental_regex_strawperson, msg);
```

Comment:
I want to write a lit test for this. With FileCheck, I can just run it twice and define a `--check-prefix`, easy. Is there a way to differentiate `expected-note{...}` based on whether a flag was passed? That is, I don't want a note if `--enable-experimental-regex` is not passed, but I want to check the note if it is.